src/frontend/data/stores/useLivePanelState/index.ts

```diff
@@ -0,0 +1,79 @@
+import create from 'zustand';
+
+export enum LivePanelDetail {
+  JOIN_DISCUSSION,
+  CHAT,
+  APPLICATION,
+}
+
+type State = {
+  /**
+   * By default, panel is not visible and doesn't support any detail
+   */
+  availableDetails: LivePanelDetail[];
+  currentDetail?: LivePanelDetail;
+  isPanelVisible: boolean;
+  /**
+   * config the store to set available details and current selection
+   * @param details all available details in the panel
+   * @param detailToSelect optional detail to select
+   * this detail must be present in { @see details }
+   * @returns void
+   */
+  setAvailableDetails: (
+    details: LivePanelDetail[],
+    detailToSelect?: LivePanelDetail,
+  ) => void;
+  /**
+   * open or close the panel and optionally change the selection in the panel
+   * @param detailToSelect detail we want to display in the panel,
+   * this detail must be available in { @see state.availableDetails }
+   * else detail selection wont change
+   * @returns void
+   */
+  setPanelVisibility: (
+    isVisible: boolean,
+    detailToSelect?: LivePanelDetail,
+  ) => void;
+};
+
+export const useLivePanelState = create<State>((set) => ({
+  availableDetails: [],
+  currentDetail: undefined,
+  isPanelVisible: false,
+  setAvailableDetails: (details, detailToSelect) =>
+    set((state) => ({
+      //  close the pane if there is nothing to display in the panel
+      isPanelVisible: state.isPanelVisible && details.length > 0,
+      //  update available details
+      availableDetails: details,
+      //  check detail to select is within available details
+      //  this is an invariant for components using the store
+      currentDetail:
+        //  if target detail is defined and available
+        detailToSelect !== undefined && details.includes(detailToSelect)
+          ? //  select it
+            detailToSelect
+          : //  else try current detail
+          state.currentDetail && details.includes(state.currentDetail)
+          ? //  select it
+            state.currentDetail
+          : //  else try available details
+          details.length > 0
+          ? //  select it
+            details[0]
```

Comment:
I don't understand why you select the first item in the list. This is an arbitrary choice ?

Reply:
Yep, this is an arbitary choice, if I can't find any valid item to select I select the first one since I'm sure there is at least one valid element to select